File: java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

public class Hamming {
	String leftStrand;
	String rightStrand;

	Hamming(String leftStrand, String rightStrand) {
		if (leftStrand.length() != rightStrand.length()) {
			throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
		}

		this.leftStrand = leftStrand;
		this.rightStrand = rightStrand;
	}

	int getHammingDistance() {
		return (int) IntStream.range(0, leftStrand.length()).filter(i -> leftStrand.charAt(i) != rightStrand.charAt(i))
				.count();
	}

}

File: csharp/hamming/Hamming.cs
using System.Linq;
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException();
        }

        return Enumerable
            .Range(0, firstStrand.Length)
            .Where(i => firstStrand[i] != secondStrand[i])
            .Count();
    }
}
